src/ripple/basics/impl/Time.cpp

```diff
@@ -18,47 +18,66 @@
 //==============================================================================
 
 #include <BeastConfig.h>
-#include <ripple/basics/Time.h>
+#include <ripple/beast/clock/chrono_util.h>
+#include <ripple/basics/chrono.h>
+#include <iomanip>
+#include <sstream>
+#include <tuple>
 
 namespace ripple {
 
-// VFALCO TODO Tidy this up into a RippleTime object
-
-//
-// Time support
-// We have our own epoch.
-//
-
-boost::posix_time::ptime ptEpoch ()
-{
-    return boost::posix_time::ptime (boost::gregorian::date (2000, boost::gregorian::Jan, 1));
-}
-
-int iToSeconds (boost::posix_time::ptime ptWhen)
+static
+std::tuple<int, unsigned, unsigned>
+civil_from_days(int z) noexcept
 {
-    return ptWhen.is_not_a_date_time ()
-           ? -1
-           : (ptWhen - ptEpoch ()).total_seconds ();
+    z += 719468;
+    const int era = (z >= 0 ? z : z - 146096) / 146097;
+    const unsigned doe = static_cast<unsigned>(z - era * 146097);          // [0, 146096]
+    const unsigned yoe = (doe - doe/1460 + doe/36524 - doe/146096) / 365;  // [0, 399]
+    const int y = static_cast<int>(yoe) + era * 400;
+    const unsigned doy = doe - (365*yoe + yoe/4 - yoe/100);                // [0, 365]
+    const unsigned mp = (5*doy + 2)/153;                                   // [0, 11]
+    const unsigned d = doy - (153*mp+2)/5 + 1;                             // [1, 31]
+    const unsigned m = mp + (mp < 10 ? 3 : -9);                            // [1, 12]
+    return std::tuple<int, unsigned, unsigned>(y + (m <= 2), m, d);
```

Comment:
I don't actually have it set up as a library.  The paper is meant to be a "How to code your date library" cookbook.

On the other hand, I do have a date lib based on these formulas that we could pull.  It does a bit more than we need, though there do exist places where we could take more advantage of it.  For example `to_string` could then be written:
````
std::string
to_string(std::chrono::system_clock::time_point tp)
{
    using namespace date;
    using namespace std::chrono;
    std::ostringstream str;
    str << floor<seconds>(tp);
    return str.str();
}
````
Here is that lib:  https://github.com/HowardHinnant/date

We would only want the file date.h from that lib.  Everything else would go unused.  We would need to get rid of our floor/ceil/round functions as they are supplied by date.h.  On gcc and clang we could do:
````
using namespace date::literals;
auto constexpr ripple_epoch = date::sys_days{2000_y/jan/1};
````
On VS we would have to remove the `constexpr`.  Then other timestamp constants we now express in seconds, such as:
````
// Fri Feb 26, 2016 9:00:00pm PST
const NetClock::time_point STAmountSO::soTime2{509864400s};
````
could become:
````
using namespace std::chrono_literals;
auto constexpr PST = -8h;
// Fri Feb 26, 2016 9:00:00pm PST
const NetClock::time_point STAmountSO::soTime2{
    date::sys_days{feb/26/2016} + 21h - PST - ripple_epoch};
````
Whether we want to do all that is an open question, and one I don't want to answer.  I will import date.h and make changes like this if desired.  I'm neutral on it (too close for an unbiased decision).